src/geom.rs

```diff
@@ -177,40 +229,27 @@ fn create_continuous_linering(linestrings: &Vec<Line>) -> Result<Line> {
     }
 
     // Convert the endpoint positions to a hashable type (tuple) and build the index map
-    let mut endpoints: HashMap<Position, Vec<usize>> = HashMap::new();
```

Comment:
Achso verstehe, du hast hier das ineffiziente Vec weggenommen un durch ein BTreeSet ersetzt, richtig?

Reply:
Die Idee war mehr, die ganzen Invarianten in einer neue Datenstruktur zu enforcen anstatt sie direkt im Algorithmus zu haben und weniger performance. Für nicht zu viele Einträge (vielleicht ein paar zehntausend für billige Datentypen wie `usize`) ist ein `Vec` eigentlich schneller als ein `BTreeSet` (das vielleicht schneller ist als ein `HashSet`), aber darum ging es hier garnicht so richtig, sondern eher um die API die ein set anbietet. Als ich gemessen habe, war die neue impl jedenfalls nicht langsamer 😅